plugins/processors/enum/enum.go

```diff
@@ -68,10 +91,10 @@ func (mapper *EnumMapper) applyMappings(metric telegraf.Metric) telegraf.Metric
 	newTags := make(map[string]string)
 
 	for _, mapping := range mapper.Mappings {
-		if mapping.Field != "" {
+		if mapping.Field != "" || len(mapping.Fields) != 0 {
 			mapper.fieldMapping(metric, mapping, newFields)
 		}
-		if mapping.Tag != "" {
+		if mapping.Tag != "" || len(mapping.Tags) != 0 {
 			mapper.tagMapping(metric, mapping, newTags)
 		}
```

Comment:
As said before, I would prefer to create a filter for tags and fields and then simply check for the filter being not-nil here. I think this will simplify the code...